src/STW.ProcessingApi.Function/Helpers/FieldConfigHelper.cs

```diff
@@ -0,0 +1,23 @@
+namespace STW.ProcessingApi.Function.Helpers;
+
+using System.Linq;
+using Newtonsoft.Json.Linq;
+
+public static class FieldConfigHelper
+{
+    public static List<string> GetValidPurposes(string fieldConfigData, string? speciesTypeName)
+    {
+        var internalMarketKey = string.IsNullOrEmpty(speciesTypeName) ? "internalMarket" : $"internalMarket-{speciesTypeName}";
+        var expression = $"$..['Purpose']..[?(@.id=='{internalMarketKey}' || @.name=='{internalMarketKey}')]..values";
+        var array = (JArray)JToken.Parse(fieldConfigData).SelectTokens(expression).FirstOrDefault() ?? [];
+        var internalMarketPurposes = new List<string>();
```

Comment:
Just a suggestion on how you can drop the if statement as the `Select` is the same as `map` in Java.

```csharp
    public static List<string> GetValidPurposes(string fieldConfigData, string? speciesTypeName)
    {
        var internalMarketKey = string.IsNullOrEmpty(speciesTypeName)
            ? "internalMarket"
            : $"internalMarket-{speciesTypeName}";

        var expression = $"$..['Purpose']..[?(@.id=='{internalMarketKey}' || @.name=='{internalMarketKey}')]..values";

        var internalMarketPurposes = JToken.Parse(fieldConfigData)
            .SelectTokens(expression)
            .FirstOrDefault()?
            .Select(x => ((JObject)x).GetValue("label")!.ToString())
            .ToList();

        return internalMarketPurposes ?? [];
    }
```